components/DownloadsComponent.tsx

```diff
@@ -0,0 +1,165 @@
+import { Box } from '@mui/material';
+import { RotatingLines } from "react-loader-spinner";
+import allFilteredAssets, {FilteredAsset} from "../utils/fetchReleases";
+import { useTheme } from '@mui/material/styles';
+
+const DownloadsComponent: React.FC  = () => {
+    const [originalData, setOriginalData] = useState<FilteredAsset[]>([]);
+    const [displayData, setDisplayData] = useState<FilteredAsset[]>([]);
+    const [loading, setLoading] = useState<boolean>(true);
+    const [error, setError] = useState<string | null>(null);
+    const [arch, setArch] = React.useState<string>('arm64');
+    const [os, setOs] = React.useState<string>('linux');
+    const [version, setVersion] = React.useState<string>('v7.4');
+    const [uniqueMinorVersions, setUniqueMinorVersions] = useState<string[]>([]);
+
+    const theme = useTheme();
+
+    const handleArch = (
+        event: React.MouseEvent<HTMLElement, MouseEvent>,
+        newAlignment: string | null,
+    ) => {
+        if (newAlignment) {
+        setArch(newAlignment);
+        }
+    };
+
+    const handleOs = (
+        event: React.MouseEvent<HTMLElement, MouseEvent>,
+        newOs: string | null,
+    ) => {
+    if (newOs !== os) {
+        setOs(newOs || '');
+        } else {
+        setOs('');
+        }
+    };
+
+    const handleVersion = (
+        event: React.MouseEvent<HTMLElement, MouseEvent>,
+        newVersion: string | null,
+    ) => {
+        if (newVersion !== version) {
+        setVersion(newVersion || '');
+        } else {
+        setVersion('');
+        }
+    }
+
+    useEffect(() => {
+        // Function to fetch release assets
+        let ignore = false;
+        const fetchAssets = async () => {
+            if (!ignore) {
+                setLoading(true);
+            try {
+                const assets = await allFilteredAssets();
+                setOriginalData(assets); // Save the original data
+                setDisplayData(assets);  // Start with all data displayed
+            } catch (e) {
+                setError('Failed to fetch release assets');
+                console.error(e);
+            } finally {
+                setLoading(false);
+            }
+            } 
+        };
+        // Call the fetch function
+        fetchAssets();
+
+        // Cleanup function to avoid setting state on unmounted component
+        return () => {
+            ignore = true;
+        };
+    }, []);
+
+    useEffect(() => {
+        const uniqueMinors = new Set<string>();
+        for (const asset of originalData) {
+        const versionComponents = asset.version.split('.');
+        if (versionComponents.length === 3) { // Ensure it's Major.Minor.Patch
+            const majorMinor = `${versionComponents[0]}.${versionComponents[1]}`;
+            uniqueMinors.add(majorMinor);
+        }
+        if (uniqueMinors.size === 4) break; // Stop after collecting 4 unique minors
+        }
+        setUniqueMinorVersions(Array.from(uniqueMinors));
```

Comment:
Use useMemo here so that you don't have to create a state element for a value that is calculated from the original data.

Reply:
Done, working smoother now!